utils/file_backend_s3_test.go

```diff
@@ -13,20 +13,20 @@ func TestCheckMandatoryS3Fields(t *testing.T) {
 	cfg := model.FileSettings{}
 
 	err := CheckMandatoryS3Fields(&cfg)
-	if err == nil || err.Message != "api.admin.test_s3.missing_s3_bucket" {
+	if err != nil && err.Message != "api.admin.test_s3.missing_s3_bucket" {
 		t.Fatal("should've failed with missing s3 bucket")
 	}
 
 	cfg.AmazonS3Bucket = "test-mm"
 	err = CheckMandatoryS3Fields(&cfg)
-	if err == nil || err.Message != "api.admin.test_s3.missing_s3_endpoint" {
-		t.Fatal("should've failed with missing s3 endpoint")
+	if err != nil {
+		t.Fatal("should've not failed")
 	}
 
-	cfg.AmazonS3Endpoint = "s3.newendpoint.com"
+	cfg.AmazonS3Endpoint = ""
 	err = CheckMandatoryS3Fields(&cfg)
-	if err == nil || err.Message != "api.admin.test_s3.missing_s3_region" {
-		t.Fatal("should've failed with missing s3 region")
+	if err == nil && cfg.AmazonS3Endpoint != "s3.amazonaws.com" {
```

Comment:
I feel like this should be `err != nil || cfg.AmazonS3Endpoint != "s3.amazonaws.com"`.

Reply:
in this case, err should not occur and the AmazonS3Endpoint should be equal to "s3..."

Reply:
done